src/qe/mbp/mbp_basic_tg.cpp

```diff
@@ -43,32 +43,106 @@ struct mbp_basic_tg::impl {
     void mark_seen(expr *t) { m_seen.mark(t); }
     bool is_seen(expr *t) { return m_seen.is_marked(t); }
 
-    //Split on all ite terms, irrespective of whether
-    //they contain variables/are c-ground
+    // Split on all ite terms, irrespective of whether
+    // they contain variables/are c-ground
     bool apply() {
-        if (!m_use_mdl) return false;
+        std::function<bool(expr *)> should_split, is_true, is_false;
+        if (!m_use_mdl) {
+            should_split = [&](expr *t) { return m_tg.has_val_in_class(t); };
+            is_true = [&](expr *t) {
+                return m_tg.has_val_in_class(t) && m_mdl.is_true(t);
+            };
+            is_false = [&](expr *t) {
+                return m_tg.has_val_in_class(t) && m_mdl.is_false(t);
+            };
+        } else {
+            should_split = [](expr *t) { return true; };
+            is_true = [&](expr *t) { return m_mdl.is_true(t); };
+            is_false = [&](expr *t) { return m_mdl.is_false(t); };
+        }
+
         expr *c, *th, *el;
         expr_ref nterm(m);
         bool progress = false;
         TRACE("mbp_tg", tout << "Iterating over terms of tg";);
         // Not resetting terms because get_terms calls resize on terms
         m_tg.get_terms(terms, false);
-        for (expr* term : terms) {
-            if (is_seen(term))
-                continue;
-            if (m.is_ite(term, c, th, el)) {
+        for (expr *term : terms) {
+            if (is_seen(term)) continue;
+            if (m.is_ite(term, c, th, el) && should_split(c)) {
                 mark_seen(term);
                 progress = true;
                 if (m_mdl.is_true(c)) {
                     m_tg.add_lit(c);
                     m_tg.add_eq(term, th);
-                }
-                else {
+                } else {
```

Comment:
Elsewhere I try to standardize code to have } and else on separate lines.

Reply:
Oh. I ran clang format with the formatter for Spacer. I will update the formal file.